### PR TITLE
feat: add Apache Beam example

### PR DESCRIPTION
**Pull request by fbiville**
Add an extensive Apache Beam example using import-spec.

Note: it does not implement all the concepts exposed in import-spec but supports all the important bits.

### CODE REVIEW

**Comment by neo4j-connectors** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>PR Title 'Add Apache Beam example': subject may not be empty, type may not be empty</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: Please include a description of your PR changes.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Commit Message 'wip: add Apache Beam example': type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test]
  </td>
    </tr>
  </tbody>
</table>


> Pull request should have a description of the underlying changes.

> Pull request title should follow [Conventional commits](https://cheatography.com/albelop/cheat-sheets/conventional-commits).
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ffc89eb1cedfda49667bbe70c964a961038a3e95
</p>

**Comment by neo4j-connectors** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: Please include a description of your PR changes.</td>
    </tr>
  </tbody>
</table>


> Pull request should have a description of the underlying changes.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a5acc4d88e4a9bcd2990329071a5d37569ff1885
</p>

File: examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+
+                    var targetOutput = sourceRecords
+                            .apply(
+                                    "[target %s] Wait for implicit dependencies".formatted(target.getName()),
+                                    Wait.on(implicitDependenciesOf(target, targetOutputs)))
+                            .setCoder(sourceRecords.getCoder())
+                            .apply(
+                                    "[target %s] Spread records across batches".formatted(target.getName()),
+                                    SpreadAcrossBatches.range(
+                                            Runtime.getRuntime().availableProcessors()))
```
Comment (ali-ince) [inline examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java:140]: I'm not sure about this at all, but could we just use one of the stock transforms for this purpose, such as:

```suggestion
                            .apply("[target %s] Assign keys to records".formatted(target.getName()),
                                    WithKeys.of((SerializableFunction<GenericRecord, Integer>) input -> ThreadLocalRandom.current().nextInt(Runtime.getRuntime().availableProcessors())))
                            .setCoder(
                                    KvCoder.of(VarIntCoder.of(), sourceRecords.getCoder()))
 ``

**Review by ali-ince — APPROVED** [review]
Looks good to me, just proposed to use `WithKeys` transform instead of the `SpreadAcrossBatches`. But that's just a nit.

File: examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java (hunk 1 above)
Comment (fbiville) [reply examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java:140]: nice, my search-fu missed it yesterday

Comment (fbiville) [reply examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java]: fixed in 98d2c86e6bb0a38091d718f672e7786aae7db97f

**Pull request merged**